From b2bcd208670b2132f8c1b2bc5ed9799e631f05f7 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Sun, 10 Oct 2021 03:40:52 -0700
Subject: [PATCH 12/22] Launcher3: Improve search bar UI in All Apps

This is similar to, but not exactly the same as, Pixel stock:

- Background color with contrast in light mode
- Round, flat search bar surface with color matching the header
  background
- Solid rounded background when positioned at the beginning of the list
- Subtle placeholder text (medium weight) without icon
- Same font size as Settings search bar

Change-Id: Ifa9fabe3a8236513fad8030c0bd8ed4d27ebd549
---
 res/drawable/all_apps_search_hint.xml                         | 2 +-
 res/drawable/bg_all_apps_searchbox.xml                        | 3 +--
 res/layout/search_container_all_apps.xml                      | 4 ++--
 res/values-v31/colors.xml                                     | 2 +-
 .../launcher3/allapps/ActivityAllAppsContainerView.java       | 1 +
 .../launcher3/allapps/search/AppsSearchContainerLayout.java   | 1 -
 6 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/res/drawable/all_apps_search_hint.xml b/res/drawable/all_apps_search_hint.xml
index b2ff7a428e..a066e98d9d 100644
--- a/res/drawable/all_apps_search_hint.xml
+++ b/res/drawable/all_apps_search_hint.xml
@@ -16,5 +16,5 @@
 -->
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
     <item android:color="@android:color/transparent" android:state_focused="true" />
-    <item android:color="?android:attr/colorAccent"/>
+    <item android:color="?android:attr/textColorSecondary" android:alpha="?android:attr/disabledAlpha" />
 </selector>
\ No newline at end of file
diff --git a/res/drawable/bg_all_apps_searchbox.xml b/res/drawable/bg_all_apps_searchbox.xml
index 3c321e4c49..b95e468c18 100644
--- a/res/drawable/bg_all_apps_searchbox.xml
+++ b/res/drawable/bg_all_apps_searchbox.xml
@@ -14,7 +14,6 @@
      limitations under the License.
 -->
 <shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="rectangle">
-    <solid android:color="?attr/popupColorPrimary" />
+    <solid android:color="?attr/allappsHeaderProtectionColor" />
     <corners android:radius="@dimen/rounded_button_radius" />
-    <stroke android:color="?attr/allappsHeaderProtectionColor" android:width=".5dp" />
 </shape>
\ No newline at end of file
diff --git a/res/layout/search_container_all_apps.xml b/res/layout/search_container_all_apps.xml
index db218c3d4b..0ad13d538b 100644
--- a/res/layout/search_container_all_apps.xml
+++ b/res/layout/search_container_all_apps.xml
@@ -22,16 +22,16 @@
     android:layout_gravity="top|center_horizontal"
     android:background="@drawable/bg_all_apps_searchbox"
     android:focusableInTouchMode="true"
-    android:gravity="center"
     android:hint="@string/all_apps_search_bar_hint"
     android:imeOptions="actionSearch|flagNoExtractUi"
     android:importantForAutofill="no"
     android:inputType="text|textNoSuggestions|textCapWords"
     android:maxLines="1"
     android:padding="8dp"
+    android:paddingStart="16dp"
     android:saveEnabled="false"
     android:scrollHorizontally="true"
     android:singleLine="true"
     android:textColor="?android:attr/textColorSecondary"
     android:textColorHint="@drawable/all_apps_search_hint"
-    android:textSize="16sp" />
\ No newline at end of file
+    android:textSize="20sp" />
diff --git a/res/values-v31/colors.xml b/res/values-v31/colors.xml
index 841e07b263..7ea90f837c 100644
--- a/res/values-v31/colors.xml
+++ b/res/values-v31/colors.xml
@@ -19,7 +19,7 @@
 <resources  xmlns:android="http://schemas.android.com/apk/res/android">
     <color name="popup_color_primary_light">@android:color/system_accent2_50</color>
     <color name="popup_color_secondary_light">@android:color/system_neutral2_100</color>
-    <color name="popup_color_tertiary_light">@android:color/system_neutral2_300</color>
+    <color name="popup_color_tertiary_light">@android:color/system_neutral2_100</color>
     <color name="popup_text_color_light">@android:color/system_neutral1_900</color>
     <color name="popup_color_neutral_dark">@android:color/system_neutral1_1000</color>
     <color name="popup_color_primary_dark">@android:color/system_neutral2_800</color>
diff --git a/src/com/android/launcher3/allapps/ActivityAllAppsContainerView.java b/src/com/android/launcher3/allapps/ActivityAllAppsContainerView.java
index 7e943a907b..b134d29f12 100644
--- a/src/com/android/launcher3/allapps/ActivityAllAppsContainerView.java
+++ b/src/com/android/launcher3/allapps/ActivityAllAppsContainerView.java
@@ -674,6 +674,7 @@ public class ActivityAllAppsContainerView<T extends Context & ActivityContext>
             mTabsProtectionAlpha = tabsAlpha;
             invalidateHeader();
         }
+        getSearchView().setBackgroundResource(R.drawable.bg_all_apps_searchbox);
         if (mSearchUiManager.getEditText() == null) {
             return;
         }
diff --git a/src/com/android/launcher3/allapps/search/AppsSearchContainerLayout.java b/src/com/android/launcher3/allapps/search/AppsSearchContainerLayout.java
index 78c305b2eb..d83efd6c16 100644
--- a/src/com/android/launcher3/allapps/search/AppsSearchContainerLayout.java
+++ b/src/com/android/launcher3/allapps/search/AppsSearchContainerLayout.java
@@ -77,7 +77,6 @@ public class AppsSearchContainerLayout extends ExtendedEditText
 
         mSearchQueryBuilder = new SpannableStringBuilder();
         Selection.setSelection(mSearchQueryBuilder, 0);
-        setHint(prefixTextWithIcon(getContext(), R.drawable.ic_allapps_search, getHint()));
 
         mContentOverlap =
                 getResources().getDimensionPixelSize(R.dimen.all_apps_search_bar_content_overlap);
-- 
2.34.1

